pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -82,6 +91,18 @@ func SetupWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
 		Complete()
 }
 
+func getPodOptions(integrationOpts map[string]any) (configapi.PodIntegrationOptions, error) {
+	opts, ok := integrationOpts[corev1.SchemeGroupVersion.WithKind("Pod").String()]
+	if !ok {
+		return configapi.PodIntegrationOptions{}, errPodOptsNotFound
```

Comment:
We could use default values instead. But this is ok too.

Reply:
I think that avoiding similar defaults would be better since this `PodIntegrationOptions` is the same as `integrations.podIntegrationOptions` in the configAPI, and configAPI has CRD defaulting here: https://github.com/kubernetes-sigs/kueue/blob/44adc22853fce5daf7f898a6bb3a17ccfb0a37bf/apis/config/v1beta1/defaults.go.

Also, when there is no "pod" key in the `options.IntegrationOptions`, that looks like a bug.